File: src/main/javascript/leetcode/P665_Non_decreasing_Array.js
var checkPossibility = function(nums) {
    for (let i = 0, len = nums.length; i < len; i++) {
        if (isSorted(nums, i)) {
            return true;
        }
    }
    return false;
};

function isSorted(nums, excludedIndex) {
    for (let i = 0, len = nums.length; i < len; i++) {
        if (i == excludedIndex) {
            continue;
        }

        let nextIndex = i + 1 === excludedIndex ? i + 2 : i + 1;
        if (nextIndex === len) {
            break;
        }

        if (nums[i] > nums[nextIndex]) {
            return false;
        }
    }
    return true;
}

console.log(checkPossibility([4,2,1]));

File: src/main/golang/leetcode/P665_Non_decreasing_Array.go
package leetcode

func checkPossibility(nums []int) bool {
	for i, len := 0, len(nums); i < len; i++ {
		if isSorted(nums, i) {
			return true
		}
	}
	return false
}

func isSorted(nums []int, excludedIndex int) bool {
	for i, len := 0, len(nums); i < len; i++ {
		if i == excludedIndex {
			continue
		}

		nextIndex := i + 1
		if nextIndex == excludedIndex {
			nextIndex = i + 2
		}

		if nextIndex >= len {
			break
		}

		if nums[i] > nums[nextIndex] {
			return false
		}
	}
	return true
}
